src/main/java/org/jenkinsci/maven/plugins/hpi/InitializeMojo.java

```diff
@@ -0,0 +1,55 @@
+package org.jenkinsci.maven.plugins.hpi;
+
+import io.jenkins.lib.versionnumber.JavaSpecificationVersion;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+
+/**
+ * Configure Maven for the desired version of Java.
+ *
+ * @author Basil Crow
+ */
+@Mojo(name = "initialize", defaultPhase = LifecyclePhase.INITIALIZE)
+public class InitializeMojo extends AbstractJenkinsMojo {
+
+    @Override
+    public void execute() throws MojoExecutionException {
+        setSurefireProperties();
+    }
+
+    private void setSurefireProperties() throws MojoExecutionException {
+        if (JavaSpecificationVersion.forCurrentJVM().isOlderThan(new JavaSpecificationVersion("9"))) {
+            // nothing to do
+            return;
+        }
+
+        String addOpens = getAddOpens();
+        if (addOpens == null) {
+            // core older than 2.339, ignore
+            return;
+        }
+
+        String argLine = project.getProperties().getProperty("argLine");
```

Comment:
This is not great—it ties us to using a property of this name in `plugin-pom`. (I am also not sure whether this works correctly when `argLine` is overridden with references to other properties. Worth checking `-Penable-jacoco` specifically.)

Better would be for this mojo to define a fresh property, `jenkins.addOpens` or whatever, and adjust `plugin-pom` to use it:

```xml
<jenkins.addOpens/>
<argLine>${jenkins.addOpens} -Xms768M -etc</argLine>
```

Reply:
> This is not great—it ties us to using a property of this name in `plugin-pom`.

And how is this not great?

> (I am also not sure whether this works correctly when argLine is overridden with references to other properties. Worth checking -Penable-jacoco specifically.)

Have you proven this, or this idle speculation?

> Better would be for this mojo to define a fresh property, `jenkins.addOpens` or whatever, and adjust `plugin-pom` to use it

commit 0e11da2

https://github.com/jenkinsci/plugin-pom/pull/524